Rematch/UI.lua

```diff
@@ -9,251 +9,93 @@ local ns = select(2, ...)
-                    else
-                        yoffset = 8
-                    end
-                    button.WinRecord:SetPoint('BOTTOMRIGHT',-3,4)
-                    button.WinRecord:SetHitRectInsets(0,0,0,-2)
-                end
-
-                if button.Preferences:IsShown() then
-                    xoffset = xoffset - button.Preferences:GetWidth()
-                end
-                if button.Notes:IsShown() then
-                    xoffset = xoffset - button.Notes:GetWidth()
-                end
-
-                script:SetPoint('RIGHT', xoffset, yoffset)
-
-                xoffset = xoffset - script:GetWidth()
-
-                if button.slim then
-                    button.Name:SetPoint('RIGHT',xoffset-2,0)
-                else
-                    button.Name:SetPoint('TOPRIGHT',xoffset-1,-6)
-                end
-            else
-                script:Hide()
-            end
-        end)
-    end
-
-    local function move(button, add)
-        if not button:IsShown() then
-            return 0
-        end
+    -- When a script is added/removed, refresh the teams list.
+    self:RegisterMessage('PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE')
+end
 
-        local point, relative, relativePoint, x, y = button:GetPoint()
-        button:SetPoint(point, relative, relativePoint, x + 21, y)
-        return add
+function Addon:PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE()
+    if Rematch.teamsPanel:IsVisible() then
+        Rematch.teamsPanel:Refresh()
     end
-
-    self:SecureHook(RematchLoadedTeamPanel, 'Update', function(panel)
-        local footnotes = panel.Footnotes
-        local script = scriptButtons[footnotes]
-
-        if self:GetScript(RematchSettings.loadedTeam) then
-            script.key = RematchSettings.loadedTeam
-            script:Show()
-            script:ClearAllPoints()
-            script:SetPoint('LEFT', 5, -0.5)
-
-            local fx = 5 + 21
-            fx = fx + move(footnotes.Preferences, 21)
-            fx = fx + move(footnotes.Notes, 21)
-            fx = fx + move(footnotes.WinRecord, footnotes.WinRecord:GetWidth())
-            fx = fx + move(footnotes.Maximize, 21)
-            fx = fx + move(footnotes.Close, 21)
-
-            local footnoteWidth = fx + 4
-            local panelWidth = panel.maxWidth or 280
-
-            footnotes:SetWidth(footnoteWidth)
-            footnotes:Show()
-            panel:SetWidth(panelWidth-footnoteWidth-3)
-        else
-            script:Hide()
-        end
-    end)
-
-    self:HookScript(RematchJournal, 'OnShow', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 870)
-        UpdateUIPanelPositions()
-    end)
-    self:HookScript(RematchJournal, 'OnHide', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 710)
-        UpdateUIPanelPositions()
-    end)
 end
 
+
 function Addon:OnDisable()
-    tDeleteItem(Rematch:GetMenu('TeamMenu'), teamMenu)
-end
+    -- TODO: hide menu, disable badge?
```

Comment:
open todo

> hide menu, disable badge?

Reply:
* I don't see an easy way to remove menu entries. 
* Badges are easy to disable with a check in the callback.

Is it best to just reload after this is disabled?

Reply:
- done for rematch 4: menu, badges, events
- done for rematch 5: events
- missing for rematch 5: menu, badges

Reply:
rematch5 menu and badges requires new api. there are two working hacks around it in #61, so done as well